draft-ietf-webpush-protocol.xml

```diff
@@ -1084,7 +1084,7 @@ HEADERS      [stream 82] +END_STREAM
             A push service MAY expire a subscription set at any time which MUST also
             expire all push message subscriptions in the set. If a user agent has an
             outstanding request to a push subscription set (<xref target="monitor-set"/>)
-            this can be signaled by returning a 400-series status code, such as 410 (Gone).
+            this can be signaled by returning a 410 (Gone) status code.

```

Comment:
Agreed. It just feels a bit odd that we're implying that the client made an
error (4xx) rather than the PS failed to send the message due to a TTL
expiring. Fun of being a service caught between two things, I suppose.
On Jun 1, 2016 1:51 PM, "Brian Raymor" <notifications@github.com> wrote:

> In draft-ietf-webpush-protocol.xml
> <https://github.com/webpush-wg/webpush-protocol/pull/92#discussion_r65440057>
> :
>
> > @@ -1084,7 +1084,7 @@ HEADERS      [stream 82] +END_STREAM
> >              A push service MAY expire a subscription set at any time which MUST also
> >              expire all push message subscriptions in the set. If a user agent has an
> >              outstanding request to a push subscription set (<xref target="monitor-set"/>)
> > -            this can be signaled by returning a 400-series status code, such as 410 (Gone).
> > +            this can be signaled by returning a 410 (Gone) status code.
>
> It's specific to gateway or proxy timeouts though:
>
> *The 504 (Gateway Timeout) status code indicates that the server, while
> acting as a gateway or proxy, did not receive a timely response from an
> upstream server it needed to access in order to complete the request.*
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/webpush-wg/webpush-protocol/pull/92/files/9749574d2c557e20828944384dde0ba510b5bb8a#r65440057>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AACLq_wYclHFSnRXM8h49oseupQaFUknks5qHfDRgaJpZM4IrI5F>
> .
>